exatomic/interfaces/xyz.py

```diff
@@ -45,19 +45,39 @@ def parse_atom(self, unit='Angstrom', names=('symbol', 'x', 'y', 'z')):
         Args:
             unit (str): Default xyz unit of length is the Angstrom
         """
+        # find where we have the number of atoms and where each
+        # frame starts
+        starts = []
+        nats = []
+        for idx, line in enumerate(self):
+            try:
+                nat = int(line.split()[0])
+                starts.append(idx)
+                nats.append(nat)
+            except ValueError:
+                continue
+        # set the rows that we want to skip
+        # this would be the number of atoms and comment
+        to_skip = np.concatenate((starts, np.array(starts)+1))
+        comments = [self[i+1] for i in starts]
         df = pd.read_csv(six.StringIO(six.u(str(self))), delim_whitespace=True,
                                       names=names, header=None,
-                                      skip_blank_lines=False)
-        # The following algorithm works for both trajectory files and single xyz files
-        nats = pd.Series(df[df[['y', 'z']].isnull().all(axis=1)].index)
-        nats = nats[nats.diff() != 1].values
-        comments = nats + 1
-        nats = df.loc[nats, 'symbol']
-        comments = df.loc[comments, :].dropna(how='all').index
-        initials = nats.index.values.astype(np.int64) + 2
-        counts = nats.values.astype(np.int64)
-        frame, _, indices = starts_counts(initials, counts)
-        df = df[df.index.isin(indices)]
```

Comment:
Can you speak to why changes to `xyz.py` are needed? Because there are no changes to the tests, it seems like the previous implementation was working correctly. If there is a gap that the above changes address, we should capture that by updating the tests to cover that edge case.

Reply:
Yeah, so there was an issue when the comment line had more than four strings separated by a space. I noticed it when it runs through the test with `'ZrO2.xyz'` where the comment line has 5 space delimited strings.

Something I looked into was using the `on_bad_lines='skip'` and we could just have the code look in the `df.loc[0,'symbol']` value for the number of atoms. I'm not sure how to then find the first non-comment line. For most of them, it works. Just that one case it does not.

I'm open to any suggestions you have on how to improve that section of the code. Mind you this is only an issue when using pandas>=1.4.0.